### DISCUSSION TITLE
Auto-Kick AFK Players

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [sakura-ryoko/afkplus#23] [Q&A] [answered]
Hello,

I didn't see this in the configuration or any of the documentation, but I'm looking for a mod that will allow me to automatically kick AFK players after a certain duration.

Is this something that exists in this mod or should I create a feature request?

**Comment by sakura-ryoko** [comment]
I'm not entirely sure why someone would want to do this...  Unless you want it to happen after X amount of time, like after an hour or some configurable amount after they are marked as AFK at a different timer.
The intention of this Mod was to simply convey when people are AFK, like on the Player List; generally speaking; but it would need to be it's own "manually enable" opt-in feature, I would assume, as Kicking players is not a very friendly approach to managing AFK, generally speaking; so I didn't invest any thought to the idea previously.

**Reply by TheBrambleShark** [reply]
Yeah, that's the intent.
If a player is afk for an hour, for example, they would be kicked.
This was a feature of Essentials (and later EssentialsX) for Bukkit/Spigot, which also handled AFK notifications on that platform. And I feel it would make sense to keep this functionality together.

I'm perfectly ok with this being an opt-in feature, especially as it's not applicable to everyone.

As an implementation detail, I think a good way to do this would be to set a default value of `0`:
```ini
[AfkPlusOptions]
# The number of seconds until an AFK user is automatically kicked. Set to 0 to disable.
automaticKickTimer = 0
```

**Reply by TheBrambleShark** [reply]
As one example of a practical use case, some host servers on services like Exaroton where hosts are charged only when the server is up and running. If someone goes afk and then goes to bed, the server could remain running all night because there is a single afk player. Automatically kicking the player after a certain amount of time being afk would allow the server to shut down automatically overnight, saving the host money.

This doesn't really apply to everyone but could be useful for small servers (or generally anyone not hosting on a VPS).

Of course, an alternative would just be to stop the server at night, but the idea of these hosts is the server is available whenever people want to use it, so if it's off having someone try to join would start it.

**Comment by sakura-ryoko** [comment]
I'll get to this in the next update, but I'll be quite busy this week with other things.

**Comment by sakura-ryoko** [comment, marked as answer]
Check Out the new release version 1.4.9.
(Not yet on Modrinth, but check Github)

https://github.com/sakura-ryoko/afkplus/releases

I also highly recommend you reset your afkplus.toml file for the proper default settings.

**Reply by sakura-ryoko** [reply]
Let me know how it works for you, and sorry that there are a ton of settings for it; but my goal is to make such features to be highly configurable; like down to the formatting of the message that players get when kicked, etc; via Placeholder API
